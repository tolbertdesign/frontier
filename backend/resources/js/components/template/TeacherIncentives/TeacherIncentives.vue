<template>
  <div class="text-center">
    <!-- Teacher incentive image for mobile -->
    <img
      :src="`https://${this.$store.state.s3Bucket}.s3.amazonaws.com/assets/images/teacher_incentives_mobile.png`"
      class="p-2 md:hidden"
    >

    <!-- Teacher incentive image for desktop -->
    <img
      :src="`https://${this.$store.state.s3Bucket}.s3.amazonaws.com/assets/images/teacher_incentives_desktop.png`"
      class="p-2 md:inline hidden"
    >
  </div>
</template>

<script>
export default {
  name: 'TeacherIncentives',
  status: 'prototype',
  release: '1.0.0',
}
</script>
