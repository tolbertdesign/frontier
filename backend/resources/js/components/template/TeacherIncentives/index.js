import TeacherIncentives from './TeacherIncentives'

export default TeacherIncentives
