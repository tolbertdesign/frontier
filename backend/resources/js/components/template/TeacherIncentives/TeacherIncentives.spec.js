import { shallowMount, createLocalVue } from '@vue/test-utils'
import TeacherIncentives from './TeacherIncentives.vue'

const localVue = createLocalVue()

describe('TeacherIncentives', () => {
  const $storeForTeacher = {
    state: {
      User: {
        is_teacher_user: true,
      },
      s3Bucket: 'AWS_Bucket',
    },
  }

  const $storeNotForTeacher = {
    state: {
      User: {
        is_teacher_user: false,
      },
      s3Bucket: 'AWS_Bucket',
    },
  }

  it('matches snapshot when user is a teacher and incentives are not hidden', () => {
    const wrapper = shallowMount(TeacherIncentives, {
      localVue,
      mocks: { $store: $storeForTeacher },
      propsData: {
        hide_teacher_incentives: false,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when user is a teacher and incentives are hidden', () => {
    const wrapper = shallowMount(TeacherIncentives, {
      localVue,
      mocks: { $store: $storeForTeacher },
      propsData: {
        hide_teacher_incentives: true,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when user is not a teacher and incentives are not hidden', () => {
    const wrapper = shallowMount(TeacherIncentives, {
      localVue,
      mocks: { $store: $storeNotForTeacher },
      propsData: {
        hide_teacher_incentives: false,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
