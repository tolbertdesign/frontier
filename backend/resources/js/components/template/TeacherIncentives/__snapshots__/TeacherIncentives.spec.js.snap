// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherIncentives matches snapshot when user is a teacher and incentives are hidden 1`] = `<div class="text-center" hide_teacher_incentives="true"><img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_mobile.png" class="p-2 md:hidden"> <img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_desktop.png" class="p-2 md:inline hidden"></div>`;

exports[`TeacherIncentives matches snapshot when user is a teacher and incentives are not hidden 1`] = `<div class="text-center"><img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_mobile.png" class="p-2 md:hidden"> <img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_desktop.png" class="p-2 md:inline hidden"></div>`;

exports[`TeacherIncentives matches snapshot when user is not a teacher and incentives are not hidden 1`] = `<div class="text-center"><img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_mobile.png" class="p-2 md:hidden"> <img src="https://AWS_Bucket.s3.amazonaws.com/assets/images/teacher_incentives_desktop.png" class="p-2 md:inline hidden"></div>`;
